File: AD-HOC/2378/main.js
function exceeded(n, c) {
    let weight = 0;
    while (n--) {
        let [s, e] = lines.shift().split(' ').map(Number);
        weight += e - s;
        if (weight > c) return true;
    }
    return false;
}

var lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
let [n, c] = lines.shift().split(' ').map(Number);
console.log(exceeded(n, c) ? 'S' : 'N');

File: AD-HOC/2378/main.dart
import 'dart:io';

bool exceeded(int n, int c) {
  int weight = 0;
  while (n-- > 0) {
    var [s, e] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
    weight += e - s;
    if (weight > c) return true;
  }
  return false;
}

void main() {
  var [n, c] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  print(exceeded(n, c) ? 'S' : 'N');
}
